pandas/core/ops.py

```diff
@@ -622,6 +622,10 @@ def _is_offset(self, arr_or_obj):
         """ check if obj or all elements of list-like is DateOffset """
         if isinstance(arr_or_obj, ABCDateOffset):
             return True
+        elif (is_datetime64_dtype(arr_or_obj) or
+              is_timedelta64_dtype(arr_or_obj)):
+            # Don't want to check elementwise for Series / array of datetime
+            return False
         elif is_list_like(arr_or_obj) and len(arr_or_obj):
```

Comment:
you actually *only* do this check if ``is_object_dtype`` is True in the first place, otherwise no point in iterating at all (if its not a scalar ``ABCDateOffset``)

Reply:
I think https://github.com/pandas-dev/pandas/pull/17980/commits/d0ea5dcb4351c76341f7d8c1785c9e789452316d has want you meant, much cleaner.